ygen/enumgen_test.go

```diff
@@ -26,69 +26,748 @@ import (
+							Name: "parent-b",
+							Parent: &yang.Container{
+								Name: "gran-gran-b",
+								Parent: &yang.Module{
+									Name: "support-module",
+								},
+							},
+						},
+					},
+					Parent: &yang.Entry{
+						Name: "parent-b",
+						Parent: &yang.Entry{
+							Name: "gran-gran-b",
+							Parent: &yang.Entry{
+								Name: "support-module",
+							},
+						},
+					},
+				},
 			},
 		},
 		wantUniqueNamesMap: map[string]string{
-			"enum-a": "Foo",
-			"enum-b": "Foo_",
+			"enum-a": "BaseModule_Foo",
+			"enum-b": "SupportModule_Foo",
+		},
+		wantUniqueNamesMapNoUnderscores: map[string]string{
+			"enum-a": "BaseModuleFoo",
+			"enum-b": "SupportModuleFoo",
+		},
+	}, {
+		name:                        "cannot resolve name clash due to camel-case lossiness and no parents to disambiguate",
+		inDefinedEnums:              map[string]bool{},
+		inDefinedEnumsNoUnderscores: map[string]bool{},
+		inNameClashSets: map[string]map[string]*yang.Entry{
+			"Foo": {
+				"enum-a": &yang.Entry{
+					Name: "enum-a",
+					Node: &yang.Enum{
+						Parent: &yang.Module{
+							Name: "base-module",
+						},
+					},
+					Parent: &yang.Entry{
+						Name: "base-module",
+					},
+				},
+				"enum-A": &yang.Entry{
+					Name: "enum-A",
+					Node: &yang.Enum{
+						Parent: &yang.Module{
+							Name: "base-module",
+						},
+					},
+					Parent: &yang.Entry{
+						Name: "base-module",
+					},
+				},
+			},
 		},
+		wantErrSubstr: "cannot resolve enumeration name clash",
 	}, {
-		name: "name clash from existing name",
+		name: "resolving name clash at grandparent for leaves and parent for typedefs",
 		inDefinedEnums: map[string]bool{
+			"Baz": true,
+			"Foo": true,
+		},
+		inDefinedEnumsNoUnderscores: map[string]bool{
+			"Baz": true,
 			"Foo": true,
 		},
-		inNameClashSets: map[string]map[string]bool{
+		inNameClashSets: map[string]map[string]*yang.Entry{
+			"Foo": {
+				"enum-a": &yang.Entry{
+					Name: "enum-a",
+					Node: &yang.Enum{
+						Parent: &yang.Container{
+							Name: "parent-a",
+							Parent: &yang.Container{
+								Name: "gran-gran-a",
+								Parent: &yang.Module{
+									Name: "base-module",
+								},
+							},
+						},
+					},
+					Parent: &yang.Entry{
+						Name: "parent-a",
+						Parent: &yang.Entry{
+							Name: "gran-gran-a",
+							Parent: &yang.Entry{
+								Name: "base-module",
+							},
+						},
+					},
+				},
+				"enum-b": &yang.Entry{
+					Name: "enum-b",
+					Node: &yang.Enum{
+						Parent: &yang.Container{
+							Name: "parent-b",
+							Parent: &yang.Container{
+								Name: "gran-gran-b",
+								Parent: &yang.Module{
+									Name: "base-module",
+								},
+							},
+						},
+					},
+					Parent: &yang.Entry{
+						Name: "parent-b",
+						Parent: &yang.Entry{
+							Name: "gran-gran-b",
+							Parent: &yang.Entry{
+								Name: "base-module",
+							},
+						},
+					},
+				},
+			},
+		},
+		wantUniqueNamesMap: map[string]string{
```

Comment:
This is where I think we need a difference in implementation. I'd expect that this is `ParentA_Foo` and `ParentB_Foo` when compressing paths is disabled. 

On this note, did we check what the diff created running this for the uncompressed example is? It should really be nothing.

Reply:
I've added test coverage for uncompressed schemas, where if any conflict occurs, generation simply fails due to enumeration leaves being named according to their entire path.

Reply:
I just checked, `uexampleoc` has no changes between master branch and the last PR #393

Reply:
Great, thanks. It looks like my comment above is outdated because `gran-gran-a` is the node that's used as the prefix here.